### PR TITLE
Updating host version to 3.20.1

### PR DESCRIPTION
**Pull request by v-imohammad**
### Issue describing the changes in this PR

resolves #issue_for_this_pr

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### CODE REVIEW

**Review by VpOfEngineering — APPROVED** [review]

File: src/Azure.Functions.Cli/ExtensionBundle/ExtensionBundleHelper.cs
Comment (khkh-ms) [inline src/Azure.Functions.Cli/ExtensionBundle/ExtensionBundleHelper.cs:36]: The build was failing, upon investigation we found that `ExtensionBundleManager` constructor now requires an additional non-optional parameter of `FunctionsHostingConfigOptions` object.

Comment (khkh-ms) [reply src/Azure.Functions.Cli/ExtensionBundle/ExtensionBundleHelper.cs:36]: We probably need approval from someone from host team before we merge this PR.

**Pull request merged**
